updater/updater_example_test.go

```diff
@@ -0,0 +1,98 @@
+package updater
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+)
+
+func someFunctionThatDownloadsAFile(ctx context.Context, url string) (string, error) {
+	// download file logic goes here
+	return "/tmp/downloads/examplefile.txt", nil
+}
+
+func someFunctionThatExtractsTheUpdateAndInstallIt(ctx context.Context, filePath string) error {
+	// extract and install update logic goes here
+	return nil
+}
+
+// ExampleUpdater shows how to use the updater package to
+// update an application manually.
+func ExampleUpdater() error {
+	conf := Config{
+		OmahaURL:        "http://test.omahaserver.com/v1/update/",
+		AppID:           "application_id",
+		Channel:         "stable",
+		InstanceID:      uuid.NewString(),
+		InstanceVersion: "0.0.1",
+	}
+
+	appUpdater, err := New(conf)
+	if err != nil {
+		return fmt.Errorf("init updater: %w", err)
+	}
+
+	ctx := context.TODO()
+
+	updateInfo, err := appUpdater.CheckForUpdates(ctx)
+	if err != nil {
+		return fmt.Errorf("error checking updates for app: %s, err: %w", conf.AppID, err)
```

Comment:
This should be more idiomatic:
```suggestion
		return fmt.Errorf("checking updates for app %q: %w", conf.AppID, err)
```